### ISSUE TITLE
GM_setValue + GM_getValue not working in Firefox 30

### ISSUE THREAD

**Issue by DebugMeNow** [scriptish/scriptish#231]
In the newest Firefox Version 30 following code is returning 'undefined'

GM_setValue('test', 'hello');
alert(GM_getValue('test'));

**Comment by ArkadiuszMichalski** [comment]
Confirm, I instal Scriptish 1.16, define @grant bust still have problem with this API's.  

"Naruszenie zasad dostępu Scriptish: nie można wywołać niebezpiecznego Okna - GM_getValue by resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/commonjs/sdk/timers.js"

This project is still being developed or the only solution is return to Greasemonkey?

**Comment by Lunatrius** [comment]
Just adding `@grant GM_getValue` and `@grant GM_setValue` fixed most of the scripts in Firefox 30. I assume the detection doesn't work or just isn't there. Not that I mind though.

**Comment by ArkadiuszMichalski** [comment]
I use @grant for every command what I use and don't get infos like GM_setValue()/GM_getValue() is 'undefined', I have access to GM_setValue()/GM_getValue() functions, but still this API's not work when invoke them, can't save and read data << get security error in console like show above.

Now take last Scriptish code from this site, make .xpi and the same result in Firefox 30.

Edit: 
This happen when I use @run-at document-idle, but when delete this line the problem disappears and API's works correct.  I check all this @run-at metadata (https://github.com/scriptish/scriptish/wiki/Manual%3A-Metadata-Block#run-at-new-in-scriptish) and only document-idle has problem.

**Comment by DebugMeNow** [comment]
I created following script and it does NOT work for me: I still get "undefined" as alert.

```
// ==UserScript==
// @name           Issue231
// @namespace      Issue231
// @description    Issue231-Debug-Playground
// @include        https://github.com/scriptish/scriptish/issues/231
// @grant GM_getValue
// @grant GM_setValue
// @run-at document-start
// ==/UserScript==

GM_setValue('test', 'hello');
alert(GM_getValue('test'));
```

**Comment by ArkadiuszMichalski** [comment]
Hmm, yep, @run-at document-start has the same problem as @run-at document-idle. Only other starting option works.

**Comment by ljani** [comment]
I'm using `@run-at document-end` and I'm getting the same problem.

**Comment by TheArkive** [comment]
For what it's worth Scriptish v1.11 allows GM_getValue to work ok as long as the 2nd parameter isn't a zero-length string.  Hope this helps.

**Comment by poochin** [comment]
At least, The combination of Firefox 30.0 + Scriptish has three problems about GM_getValue.

We should change to,
1. Firefox 29 + Scriptish 0.1.11+ (to downgrade Firefox)
2. Firefox 30 + Scriptish 0.1.12+ (to upgrade Scriptish)

If you select (2.), you must reinstall userscript which has @grant meta-datablock.

It is to resolve GM_getValue fails.

---

The list of problems.
1. Firefox 30.0 + Scriptish 0.1.11: GM_getValue avoid 2nd parameter (default value).
2. Firefox 30.0 + Scriptish 0.1.12: Set @run-at document-start, fail to use GM_{set,get}Value.
3. Firefox 30.0 + Scriptish 0.1.12: When to call GM_setValue, we catch error "ReferenceError: GM_getValue is not defined".

resolves.

about (1.):
  It was resoleved on Scriptish 0.1.12. install Scriptish 0.1.12 .

about (2.):
  They says on this thread, using @run-at document-end insted of document-start.

about (3.):
  To write detail to bottom.

This problem occurs to upgrad Scriptish to 0.1.12 from lower version.

I installed Firefox 30.0 + Scriptish 0.1.11, and installed userscript(it has @grant meta-detablock).
And then I upgraded Scriptish to 0.1.12.
Then the userscript fails to call @grant functions(ex: GM_getValue).

Scriptish has userscript settings on ~/.mozilla/firefox/_.default/scriptish_scripts/scriptish-config.json (case of Linux).
If we install userscript on Scriptish 0.1.11, scriptish doesn't collect @grant infomation and set to scriptish-config.json .
Then I upgraded Scriptish to 0.1.12, scriptish-config.json stil doesn't have userscript's @grant informations.
It is one of the reason that userscript can't call GM__ on Scriptish 0.1.12 .

In this case, we should reinstall userscript after upgrading Scriptish to 0.1.12 .
